File: api/src/models/Account.js
import mongoose from 'mongoose';

const accountSchema = new mongoose.Schema(
  {
    accountType: { type: String, enum: ['customer', 'supplier'], required: true },
    code: { type: String, required: true, unique: true, trim: true, uppercase: true },
    name: { type: String, required: true, trim: true },
    contactPerson: String,
    designation: String,
    email: String,
    city: String,
    state: String,
    country: String,
    pincode: String,
    address: String,
    mobileNo: String,
    contactNo: String,
    fax: String,
    cstNo: String,
    tinNo: String,
    panNo: String,
    gstNo: String,
    exciseNo: String,
    creditLimit: { type: Number, default: 0 },
    creditDays: { type: Number, default: 0 },
    openingBalance: { type: Number, default: 0 },
    openingBalanceType: { type: String, enum: ['debit', 'credit'], default: 'debit' },
    customerType: String,
    annualTurnover: String,
    sourceEmployee: String,
    activeStatus: { type: Boolean, default: true },
    gstExempt: { type: Boolean, default: false },
    billFormatAssignments: {
      type: mongoose.Schema.Types.Mixed,
      default: () => ({})
    }
  },
  { timestamps: true }
);

export const Account = mongoose.model('Account', accountSchema);

File: ims-api-net/src/Ims.Domain/Masters/Account.cs
using Ims.Domain.Common;

namespace Ims.Domain.Masters;

public sealed class Account : EntityBase, IYearScoped
{
    public string YearDatabaseName { get; set; } = string.Empty;

    public string AccountType { get; set; } = "customer";

    public string Code { get; set; } = string.Empty;

    public string Name { get; set; } = string.Empty;

    public string? ContactPerson { get; set; }

    public string? Address { get; set; }

    public string? City { get; set; }

    public string? State { get; set; }

    public string? Pincode { get; set; }

    public string? Phone { get; set; }

    public string? MobileNo { get; set; }

    public string? Email { get; set; }

    public string? Gstin { get; set; }

    public string? Pan { get; set; }

    public decimal CreditLimit { get; set; }

    public int CreditDays { get; set; }

    public decimal OpeningBalance { get; set; }

    public string OpeningBalanceType { get; set; } = "Dr";

    public string? CustomerType { get; set; }

    public bool ActiveStatus { get; set; } = true;
}
